### PR TITLE
Support ToastProgressUpdate and add unittests to validate the functionality

### PR DESCRIPTION
**Pull request by sharath2727**
Support ToastProgressUpdate and add unittests to validate the functionality

### CODE REVIEW

File: test/TestApps/ToastNotificationsTestApp/main.cpp [hunk 1]
```diff
@@ -415,7 +524,17 @@ std::map<std::string, bool(*)()> const& GetSwitchMapping()
...
+        { "VerifyUpdateNotificationDataWithNullTagAndValidGroup_Unpackaged", &VerifyUpdateNotificationDataWithNullTagAndValidGroup },
+        { "VerifyFailedUpdateNotificationDataWithNonExistentTagAndGroup_Unpackaged", &VerifyFailedUpdateNotificationDataWithNonExistentTagAndGroup },
+        { "VerifyFailedUpdateNotificationDataWithoutPostToast_Unpackaged", &VerifyFailedUpdateNotificationDataWithoutPostToast },
+        { "VerifyUpdateToastProgressDataUsingValidTagAndValidGroup", &VerifyUpdateToastProgressDataUsingValidTagAndValidGroup },
+         { "VerifyUpdateToastProgressDataUsingValidTagAndNullGroup", &VerifyUpdateToastProgressDataUsingValidTagAndNullGroup },
+        { "VerifyUpdateNotificationDataWithNullTagAndValidGroup", &VerifyUpdateNotificationDataWithNullTagAndValidGroup },
+        { "VerifyFailedUpdateNotificationDataWithNonExistentTagAndGroup", &VerifyFailedUpdateNotificationDataWithNonExistentTagAndGroup },
+        { "VerifyFailedUpdateNotificationDataWithoutPostToast", &VerifyFailedUpdateNotificationDataWithoutPostToast },
+      };
```
Comment (sharath2727) [inline test/TestApps/ToastNotificationsTestApp/main.cpp:537]: @danielayala94 @pmpurifoy There is a piece of code in main() in this file:

 if (Test::AppModel::IsPackagedProcess())
    {
        auto activationInfo = winrt::ToastActivationInfo::CreateFromActivationGuid(c_toastComServerId);
        winrt::ToastNotificationManager::Default().RegisterActivator(activationInfo);
    }

Do you know why this change was introduced? It renders someof the unittests useless. Let me know. Thanks.

File: dev/ToastNotifications/ToastNotificationManager.cpp [hunk 2]
```diff
@@ -86,13 +94,40 @@ namespace winrt::Microsoft::Windows::ToastNotifications::implementation
...
     }
-    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData /* data */, hstring /* tag */, hstring /* group */)
+    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData data, hstring tag, hstring group)
     {
-        throw hresult_not_implemented();
+        std::wstring appId{ RetrieveAppId() };
+
+        auto hr = ToastNotifications_UpdateNotificationData(appId.c_str(), tag.c_str(), group.c_str(), reinterpret_cast<ABI::IToastProgressData*>(winrt::get_abi(data.as<winrt::IToastProgressData>())));
+
+        if (hr == S_OK)
```
Comment (danielayala94) [inline dev/ToastNotifications/ToastNotificationManager.cpp:103]: nit:
```suggestion
        if (SUCCEEDED(hr))
```

File: test/TestApps/ToastNotificationsTestApp/main.cpp [hunk 3]
```diff
@@ -390,6 +387,118 @@ std::string unitTestNameFromLaunchArguments(const winrt::ILaunchActivatedEventAr
     return unitTestName;
 }
 
+constexpr auto timeout{ std::chrono::seconds(5) };
+
+winrt::ToastProgressResult progressResultOperationHelper(winrt::IAsyncOperation<winrt::ToastProgressResult> progressResultOperation)
```
Comment (danielayala94) [inline test/TestApps/ToastNotificationsTestApp/main.cpp:392]: nit:
```suggestion
winrt::ToastProgressResult ProgressResultOperationHelper(winrt::IAsyncOperation<winrt::ToastProgressResult> progressResultOperation)
```

File: test/TestApps/ToastNotificationsTestApp/main.cpp [hunk 4]
```diff
@@ -390,6 +387,118 @@ std::string unitTestNameFromLaunchArguments(const winrt::ILaunchActivatedEventAr
...
+    progressData.Status(L"SomeStatus");
+    progressData.Title(L"SomeTitle");
+    progressData.Value(0.14);
+    progressData.ValueStringOverride(L"14%");
+
+    // TODO: PoastToast 
+    auto progressResultOperation = winrt::ToastNotificationManager::Default().UpdateToastProgressDataAsync(progressData, L"Tag", L"Group");
+    auto progressResult = progressResultOperationHelper(progressResultOperation);
+
+    if (progressResult == winrt::ToastProgressResult::Succeeded)
```
Comment (danielayala94) [inline test/TestApps/ToastNotificationsTestApp/main.cpp:415]: You can condense the last lines into:

`return progressResult == winrt::ToastProgressResult::Succeeded;`

Comment (danielayala94) [reply test/TestApps/ToastNotificationsTestApp/main.cpp:415]: Same comment applies to other tests.

**Comment by sharath2727** [comment]
Will be raising a PR against WNP_ToastNotifications_L1 branch for the same.

**Pull request closed without merging**